src/lib.rs

```diff
@@ -1,3 +1,75 @@
+//!
+//! This crate provides a safe abstraction for the Landlock system calls, along with some helpers.
+//!
+//! # Use cases
+//!
+//! This crate is especially useful for build-in application sandboxing:
+//! * Parsers hardening (e.g., archive tools, file format conversion, renderers).
+//! * (Part of) applications with limited file renaming or linking needs
+//!   (e.g., some system or network services).
+//! * Monolith applications dealing with different levels of confidentiality
+//!   (e.g., web browser, email server).
+//!
+//! # Examples
+//!
+//! Examples can be found with the [`Ruleset` documentation](Ruleset)
+//! and the [sandboxer example](https://github.com/landlock-lsm/rust-landlock/blob/master/examples/sandboxer.rs).
+//!
+//! # Current limitations
+//!
+//! This crate handles the Landlock features available since Linux 5.13
+//! and then inherits some [kernel limitations](https://www.kernel.org/doc/html/latest/userspace-api/landlock.html#current-limitations)
+//! that will be addressed with future kernel releases:
+//! * File reparenting: renaming or linking a file to a different parent directory is always denied.
+//! * Filesystem topology modification: arbitrary mounts are always denied.
+//!
+//! # Compatibility
+//!
+//! This crate is designed to enable security as much as possible on supported systems.
+//! Compatibility should only be about the system supporting Landlock,
+//! not about the required action dependencies (e.g., opening a file to add it to a rule).
+//! This enables to check that the use of Landlock would be OK if supported,
+//! instead of ignoring files that should be used in a Landlock ruleset.
+//! When running the same program on different environments,
+//! this helps catch issues with the assumed context
+//! (e.g., whether a directory hierarchy exists or not).
+//!
+//! To make it easier to migrate to a new version of this library,
+//! we use the builder pattern
+//! and designed objects to require the minimal set of method arguments.
+//! Most `enum` are marked as `non_exhaustive` to enable backward-compatible evolutions.
+//!
+//! A simple approach would have been to expose the Landlock [ABI] version
+//! and ask developers to explicitly manage the related implications.
+//! This would require developers to grasp all the potential behavior changes over time,
+//! which will eventually lead to issues.
+//! Instead, we ask developers to identify optional ([best-effort](Compatible::set_best_effort))
+//! or required features.
+//! This makes more sense because this directly relies on the access-control semantic.
+//! It makes the library more difficult to implement but simpler to use.
+//! This also makes it easier to audit for most complex requirements
+//! because access rights are explicitly set as required,
+//! instead of relying on manual ABI version mapping.
+//!
+//! ## Test strategy
+//!
+//! Developers should test their sandboxed applications with a compatible kernel
```

Comment:
"compatible kernel" is vague - they are all "compatible" in some sense, in best effort mode. Maybe it would be better to say more specifically "a kernel that supports all requested Landlock features".

Further up, you are saying "This crate handles the Landlock features available since Linux 5.13" - maybe it would be easiest to refer to that section, as that seems to be the place where you have documented what set of Landlock features are supported by this version of the library?

Reply:
The supported features and their documentation should comes from the ABI enum used, for example `AccessFs::from_all(ABI::V1)`. I need to expand that and link to it.